eg/Classes/PluginInstall.py

```diff
@@ -146,38 +243,78 @@ def Import(self, filepath):
                 if os.path.isdir(path):
                     basePath = path
                     break
-            dialog = PluginOverviewDialog(
-                title="Install EventGhost Plugin",
-                pluginData=pluginData,
-                basePath=basePath,
-                message=(
-                    "Do you really want to install this plugin into "
-                    "EventGhost?"
-                )
-            )
-            result = dialog.ShowModal()
-            dialog.Destroy()
-            if result == wx.ID_CANCEL:
-                return
+
             guid = pluginData['guid']
             if guid in eg.pluginManager.database:
                 # a plugin with same GUID already exists
                 info = eg.pluginManager.database[guid]
-                if info.path.lower().startswith(eg.localPluginDir.lower()):
-                    # plugin with same GUID exists in user dir, so delete
-                    # the folder first
+                name = info.englishName
+                mData = (name, pluginData['name'])
+                if info.englishName != pluginData['name']:
+                    title = Text.replaceTitle % info.englishName
+                    message = Text.replaceMessage % mData
+
+                elif info.version != pluginData['version']:
+                    mData = (name, info.version, pluginData['version'])
+                    if info.version < pluginData['version']:
+                        title = Text.upgradeTitle % name
+                        message = Text.upgradeMessage % mData
+                    else:
+                        title = Text.downgradeTitle % name
+                        message = Text.downgradeMessage % messageData
+                else:
+                    title = Text.overwriteTitle % name
+                    message = Text.overwriteMessage % mData
+
+                dialog = PluginOverviewDialog(
+                    title=title,
+                    message=message,
+                    pluginData=pluginData,
+                    basePath=basePath,
+                )
+
+                result = dialog.ShowModal()
+                dialog.Destroy()
+                if result == wx.ID_CANCEL:
+                    return
+
+                self.BackupPlugin(info)
+                try:
                     shutil.rmtree(info.path, False)
-            dstDir = os.path.join(eg.localPluginDir, os.path.basename(basePath))
+                except WindowsError:
+                    if info.path.startswith(eg.corePluginDir):
+                        eg.PrintNotice(Text.pluginRemoveError + info.path)
+                    else:
+                        raise
```

Comment:
Maybe instead of (re-)raising an exception show a dialog that the removal of the plugin failed and the (new/updated) plugin won't be installed. (could it even be that the plugin might now be unusable?)